config.go

```diff
@@ -70,7 +69,7 @@ func initConfig() error {
 	}
 
 	if err := flags.IniParse(cfg.ConfigPath, &cfg); err != nil {
-		fmt.Println("Could not find config file:", err)
+		return err
```

Comment:
The bot should be able to start without a config file imo. Therefore I would prefer printing the log message over returning the error

Reply:
First the error "Could not find config file:" may be wrong. This is not the only reason why IniParse may return an error (As I have seen myself when setting things up).

Secon, this code (initConfig) is called from main() and in case of an error there is already a logging of the error. There is simply no need to print the error two times.

```
	if err := initConfig(); err != nil {
		printErrorAndExit("Could not initialize config:", err)
	}
```

In general, it is better to return the error to the level that handles it where it should also be printed.

Reply:
Still. The bot should start when there is an error because the file doesn't exist. Catching and printing that specific error and returning `err` otherwise would be best imo.

Reply:
If this is your intention, add the code as needed.